core/src/main/java/org/apache/iceberg/rest/RESTCatalog.java

```diff
@@ -55,7 +59,50 @@ public class RESTCatalog
   public RESTCatalog() {
     this(
         SessionCatalog.SessionContext.createEmpty(),
-        config -> HTTPClient.builder(config).uri(config.get(CatalogProperties.URI)).build());
+        config -> {
+          HTTPClient.Builder builder =
+              HTTPClient.builder(config).uri(config.get(CatalogProperties.URI));
+
+          String proxyHostname =
+              PropertyUtil.propertyAsString(
+                  config,
+                  CatalogProperties.PROXY_HOSTNAME,
+                  CatalogProperties.PROXY_HOSTNAME_DEFAULT);
+
+          int proxyPort =
+              PropertyUtil.propertyAsInt(
+                  config, CatalogProperties.PROXY_PORT, CatalogProperties.PROXY_PORT_DEFAULT);
+
+          if (!proxyHostname.isEmpty()) {
+
+            builder.withProxy(proxyHostname, proxyPort);
+
+            String proxyUsername =
+                PropertyUtil.propertyAsString(
+                    config,
+                    CatalogProperties.PROXY_USERNAME,
+                    CatalogProperties.PROXY_USERNAME_DEFAULT);
+
+            String proxyPassword =
+                PropertyUtil.propertyAsString(
+                    config,
+                    CatalogProperties.PROXY_PASSWORD,
+                    CatalogProperties.PROXY_PASSWORD_DEFAULT);
```

Comment:
same as for the others, the default should be `null` instead of specifying an empty string